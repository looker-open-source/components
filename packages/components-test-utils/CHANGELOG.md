# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.4](https://github.com/looker-open-source/components/compare/v0.10.3...v0.10.4) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.3](https://github.com/looker-open-source/components/compare/v0.10.2...v0.10.3) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.2](https://github.com/looker-open-source/components/compare/v0.10.1...v0.10.2) (2021-01-22)


### Bug Fixes

* On build properly set package.json "files" key ([#1806](https://github.com/looker-open-source/components/issues/1806)) ([c81c4bb](https://github.com/looker-open-source/components/commit/c81c4bb625c58ede49957aad90ac8d9d7b2c4b79))





## [0.10.1](https://github.com/looker-open-source/components/compare/v0.10.0...v0.10.1) (2021-01-22)


### Bug Fixes

* Properly build ems/cjs/ts for easy deep-imports ([#1804](https://github.com/looker-open-source/components/issues/1804)) ([2a523cd](https://github.com/looker-open-source/components/commit/2a523cd70b079944376c6dc87c18202e05a97b01))





# [0.10.0](https://github.com/looker-open-source/components/compare/v0.9.29...v0.10.0) (2021-01-19)


### Bug Fixes

* Properly configure build for tree-shaking support ([9202b72](https://github.com/looker-open-source/components/commit/9202b72bc5fb99eb3e1af7d9e56f3dc15b1df2a4))





## [0.9.15]

### Changed

- `@looker/components-test-utils` helpers will no longer produce `globalStyles`
