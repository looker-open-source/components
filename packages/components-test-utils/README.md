# Test Utilities

Shared functions used for mocking and testing Looker UI Components.
