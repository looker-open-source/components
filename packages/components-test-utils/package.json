{
  "name": "@looker/components-test-utils",
  "license": "MIT",
  "version": "0.9.12",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/looker-open-source/components",
    "directory": "packages/components-test-utils"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@looker/components-providers": "^0.9.12",
    "@testing-library/react": "^10.4.8",
    "@types/enzyme": "^3.10.5",
    "@types/jest": "^26.0.10",
    "@types/react": "^16.9.49",
    "@types/react-test-renderer": "^16.9.3",
    "enzyme": "^3.11.0",
    "jest-styled-components": "^6.3.4",
    "react": "^16.13.1",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^4.4.1"
  },
  "peerDependencies": {
    "react": "^16.11"
  },
  "gitHead": "62febc162e3e45124f403e2c84fc741f68fe6714"
}
