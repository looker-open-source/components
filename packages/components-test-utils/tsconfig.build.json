{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./lib"
  },
  "include": ["../../types", "src/**/*"]
}
