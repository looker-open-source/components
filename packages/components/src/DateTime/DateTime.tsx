/*

 MIT License

 Copyright (c) 2019 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React, { FC } from 'react'

type FormatStyles = 'short' | 'medium' | 'long' | 'full'

export interface DateTimeProps extends Intl.DateTimeFormatOptions {
  children?: Date
  dateStyle?: FormatStyles
  hideDate?: boolean
  hideTime?: boolean
  locale?: string
  style?: FormatStyles
  timeStyle?: FormatStyles
  timeZone?: string
}

export interface ExtendedDateTimeFormatOptions
  extends Intl.DateTimeFormatOptions {
  dateStyle?: FormatStyles
  timeStyle?: FormatStyles
}

export const DateTime: FC<DateTimeProps> = ({
  children = new Date(Date.now()),
  dateStyle,
  hideDate,
  hideTime,
  locale,
  style,
  timeStyle,
  timeZone,
}) => {
  const options: ExtendedDateTimeFormatOptions = {
    dateStyle,
    timeStyle,
    timeZone,
  }

  options.dateStyle = hideDate ? undefined : style
  options.timeStyle = hideTime ? undefined : style

  if (timeZone) {
    options.dateStyle = 'long'
    options.timeStyle = 'long'
  }

  const format = children.toLocaleDateString(locale, options)

  return <>{format}</>
}

DateTime.defaultProps = {
  style: 'medium',
}
