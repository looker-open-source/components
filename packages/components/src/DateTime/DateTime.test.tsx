import { assertSnapshot } from '@looker/components-test-utils'
import React from 'react'
import { DateTime } from './DateTime'

test('DateTime renders', () => {
  assertSnapshot(<DateTime />)
})

test('DateTime renders only date', () => {
  assertSnapshot(<DateTime hideDate />)
})

test('DateTime renders only time', () => {
  assertSnapshot(<DateTime hideTime />)
})

test('DateTime renders when passing specific Date as children', () => {
  assertSnapshot(<DateTime>{new Date(Date.UTC(2011, 1, 3, 5, 6, 2))}</DateTime>)
})

test('DateTime renders when passing specific locale', () => {
  assertSnapshot(<DateTime locale="ar-EG" />)
})

test('DateTime displays timeZone if one is passed', () => {
  assertSnapshot(<DateTime timeZone="Asia/Kolkata" />)
})

test('DateTime renders if style is passed', () => {
  assertSnapshot(<DateTime style="short" />)
})

test('DateTime renders if style is passed', () => {
  assertSnapshot(<DateTime style="long" />)
})

test('DateTime renders if style is passed', () => {
  assertSnapshot(<DateTime style="full" />)
})
