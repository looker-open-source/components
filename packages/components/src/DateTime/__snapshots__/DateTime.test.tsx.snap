// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTime displays timeZone if one is passed 1`] = `"January 30, 2020 at 3:28:01 AM GMT+5:30"`;

exports[`DateTime renders 1`] = `"Jan 29, 2020, 1:58:01 PM"`;

exports[`DateTime renders if style is passed 1`] = `"1/29/20, 1:58 PM"`;

exports[`DateTime renders if style is passed 2`] = `"January 29, 2020 at 1:58:01 PM PST"`;

exports[`DateTime renders if style is passed 3`] = `"Wednesday, January 29, 2020 at 1:58:01 PM Pacific Standard Time"`;

exports[`DateTime renders only date 1`] = `"1:58:01 PM"`;

exports[`DateTime renders only time 1`] = `"Jan 29, 2020"`;

exports[`DateTime renders when passing specific Date as children 1`] = `"Feb 2, 2011, 9:06:02 PM"`;

exports[`DateTime renders when passing specific locale 1`] = `"Jan 29, 2020, 1:58:01 PM"`;
